Update 08.outlook.md (#141)
jaybee84/ml-in-rd: content/08.outlook.md

We believe that the same challenges that are currently outlined as major pitfalls in applying ML to rare disease can be great opportunities for data generation as well as method development in moving the field forward.
During our journey through the various challenges, we identified two major areas where mindful strategies can immeasurably enhance the power of machine learning in rare disease and move the field forward.

_Emphasis on not just "more n" but "more meaningful n"._

Mindful addition of data is key for powering the next generation of analysis in rare disease data.
While there are many techniques to collate rare data from different sources, incorrect data generation may hurt the end goal even if it adds to the size of the dataset.
In our experience collaboration with domain experts have proved to be critical in gaining insight into potential sources of variation in the datasets.
As an example, in a study of neurofibromatosis type 1 (NF1) some samples were generated using vastly different surgical techniques (laser ablation and excision vs standard excision). [@doi:10.3390/genes11020226] 

> This resulted in surfacing of substantial biological differences that are a consequence of process (e.g. activation of heat shock protein related pathways), not disease related biology. 

Reviewer: We didn't specifically interrogate this in this paper, I think, or am I misremembering?

Reviewer: e.g. I think we had this hypothesis but didn't follow this particular thread (primarily because we don't have non-laser resected cNF tumor expression data to explore this difference). I think if someone looked for this in the cited paper, they wouldn't find it...but again, could be misremembering.

Author: You are correct... we had this hypothesis but did not interrogate it systematically in the paper. As I am re-reading this, I see that my phrasing suggests more than what I intended to say. Will rephrase :)

Prior collaboration with domain experts would have helped minimize these differences and generate a more robust dataset.
A few such collaborations are beginning to form and show promise in generating valuable datasets for future use.[@https://www.hopkinsmedicine.org/kimmel_cancer_center/cancers_we_treat/pediatric_oncology/research_and_clinical_trials/pratilas/nf1_biospecimen_repository.html]

Rare disease not only lacks samples, there is also a great dearth of comprehensive phenotypic-genotypic databases. 
With the ubiquity of sequencing platforms, genomic data has been easy to gather for rare disease patients.

> This is augmented by the fact that most rare diseases seem to be genetic in origin.

Reviewer 2: Citation? May be covered by `[TODO: add Refs]` below 😄

Author: :)

An important next step would be to integrate genomic and clinical data to produce comprehensive phenotypic-genotypic databases that can fuel interpretation of features extracted using ML methods.
[TODO: add Refs]

Finally, mindful sharing of data with proper metadata and attribution to enable prompt data reuse is of utmost important in building datasets that can be of great value in rare disease. [@https://www.nature.com/articles/s41576-020-0257-5]
[TODO: expand on this]

_Development of methods that reliably support mechanistic interrogation of specific rare diseases_

> A pattern that emerged readily in our literature search of application of ML methods in rare disease is that most ML related strategies were focused on classification of a patient as a rare disease patient among other non-rare disease patients.

Reviewer 2: I am hesitant to talk about a literature search for "this is not a systematic review" reasons.

### Outlook

> Throughout this perspective we have highlighted various challenges in applying ML methods to rare disease data as well as examples of approaches that address these challenges.

Reviewer: ```suggestion
Throughout this perspective, we have highlighted various challenges in applying ML methods to rare disease data as well as examples of approaches that address these challenges.
```

> Scarcity of samples while significant is not the only roadblock towards application of ML in rare disease data.

Reviewer: ```suggestion
Scarcity of samples, while significant, is not the only roadblock towards application of ML in rare disease data.
```

> The high dimensionality of the data collected using modern techniques necessitates the management of the curse of dimensionality in creative ways by learning low dimensional representations from the data.

Reviewer: ```suggestion
The high dimensionality of modern data requires creative approaches, such as learning new representations of the data, to manage the curse of dimensionality.
```

> It further requires leveraging prior knowledge and transfer learning instead of using traditional machine learning methods for appropriate interpretation of data. 

Reviewer: ```suggestion
It further requires leveraging prior knowledge and transfer learning methods to appropriately interpret data. 
```

> Additionally, anyone applying machine learning methods on rare disease data is advised to delve into fundamental techniques that increase confidence and penalize complexity of the resultant models to enhance generalizabilty of ML in rare disease.

Reviewer: I'm not sure what the point is here - little confused!

Author: This was to recap the "managing model complexity" section of the article. 
This sentence hopes to remind the reader of fundamental techniques like bootstrap and regularization ... but as is evident may need rephrasing :)

Reviewer: Made a suggested change to reflect this/improve clarity!

All of the approaches highlighted in this perspective come with certain challenges or inadequacies that breed mistrust in using these powerful techniques in rare disease.
We believe that the same challenges that are currently considered major pitfalls in applying ML to rare disease can be great opportunities for data generation as well as method development in moving the field forward.
During our journey through the various challenges, we identified two major areas where mindful strategies can immeasurably enhance the power of machine learning in rare disease and move the field forward.

> _Emphasis on not just "more n" but "more meaningful n"._

Reviewer: Just checking is this a TODO or a section header? (If a section header, suggest removing the period)

> _Emphasis on not just "more n" but "more meaningful n"._

Author: ```suggestion
_Emphasis on not just "more n" but "more meaningful n"_
```

Mindful addition of data is key for powering the next generation of analysis in rare disease data.
While there are many techniques to collate rare data from different sources, incorrect data generation may hurt the end goal even if it adds to the size of the dataset.
In our experience collaboration with domain experts have proved to be critical in gaining insight into potential sources of variation in the datasets.
As an example, an neurofibromatosis type 1 (NF1) dataset was found to contain samples collected using vastly different surgical techniques (laser ablation and excision vs standard excision). [@doi:10.3390/genes11020226] 

> While the integrative analysis in the study using transfer learning techniques was able to minimize technique related signals [@doi:10.1016/j.cels.2019.04.003], a more traditional analysis may have resulted in surfacing of substantial biological differences that are a consequence of process (e.g. activation of heat shock protein related pathways), not disease related biology. 

Reviewer: "While the integrative analysis in the study using transfer learning techniques was able to minimize technique related signals" 

I don't think that we really tested this, hesitant to make this claim.

Reviewer: After we merge, I can tackle this section - I have some ideas but a little too much rewriting to do through Github PR suggestions :)

Author: This is based on some of the discussion during writing of the earlier manuscript that since the latent variable generation finds patterns that are consistent among datasets that have divergent technical differences in them,  the result is enrichment of LVs that are biologically relevant (since the technical differences would presumably be orthogonal in different datasets). 
But maybe I am misremembering... Tagging @cgreene and @jaclyn-taroni for more insight

Author: Also.. happy to have you tackle this section after merging :)

Reviewer: Alrighty, filed as #143!

Such instances underline the fact that continuous collaboration with domain experts is needed to generate robust datasets in the future.

> A few such collaborations are beginning to show promise in generating valuable datasets for future use.[@https://www.hopkinsmedicine.org/kimmel_cancer_center/cancers_we_treat/pediatric_oncology/research_and_clinical_trials/pratilas/nf1_biospecimen_repository.html]

Reviewer: Can we cite Sci Data paper instead? Too much risk of this URL going dead. 
```suggestion
A few such collaborations are beginning to show promise in generating valuable datasets for future use.[@doi:10.1038/s41597-020-0508-5]

```

Reviewer: I do wonder if this sentence is laying on the NF focus a bit too much, but leave that to other reviewers to determine (i'm biased)

Author: maybe we can try to find other rare disease dataset releases to cite here

In addition to sample scarcity, there is a dearth of comprehensive phenotypic-genotypic databases in rare disease.

> With the ubiquity of sequencing platforms, genomic data has been easy to gather for rare disease patients.[@doi:10.1038/nrg3555; @doi:10.1038/nrg.2017.116; @doi:10.1056/NEJMra1711801]

Reviewer: ```suggestion
With the ubiquity of sequencing platforms, genomic data has been, relatively speaking, easy to gather for rare disease patients.[@doi:10.1038/nrg3555; @doi:10.1038/nrg.2017.116; @doi:10.1056/NEJMra1711801]
```

> An important next step would be to integrate genomic and clinical data to produce comprehensive phenotypic-genotypic databases that can fuel interpretation of features extracted using ML methods.

Reviewer: ```suggestion
An important next step is to develop comprehensive comprehensive genomics-driven genotype-phenotype databases that can fuel interpretation of features extracted using ML methods.
```

> Finally, mindful sharing of data with proper metadata and attribution to enable prompt data reuse is of utmost important in building datasets that can be of great value in rare disease. [@https://www.nature.com/articles/s41576-020-0257-5]

Reviewer: I think @cgreene could review and expand on this in his round-robin if it seems necessary!

_Development of methods that reliably support mechanistic interrogation of specific rare diseases_

> A pattern that emerged readily in our quest for examples of application of ML methods in rare disease is that most ML related strategies were focused on classification of a patient as a rare disease patient among other non-rare disease patients.

Reviewer: ```suggestion
The majority of ML methods for rare disease that we have investigated are applied to classification tasks. 
```

> We found that examples of methodologies interrogating mechanistic details of a specific rare disease were particularly lacking.

Reviewer: ```suggestion
Conversely, we've found few examples of methodologies that interrogate biological mechanisms of rare diseases. 
```

> Reasons for this include not only the lack of large datasets, but also lack of reliability or interpretability of the results from such analyses.

Reviewer: ```suggestion
This is likely a consequence of a dearth of methods that can tolerate the constraints imposed by rare disease research such as phenotypic heterogeneity and limited data.
```

> We believe that an intentional push towards developing methods or analytical workflows that address this would be critical in encouraging powerful analyses using rare disease data.

Reviewer: ```suggestion
An intentional push towards developing methods or analytical workflows that address this will be critical to apply machine learning approaches to rare disease data.
```

Method development with rare disease applications in mind requires the developers to bear the responsibility of ensuring that the resulting model is _trustworthy_.
The field of natural language processing has a few examples of how this can be achieved.[@https://www.aclweb.org/anthology/N16-3020.pdf, @https://www.aclweb.org/anthology/P19-1073.pdf]

> One way to increase user's trust on a developed model is by helping the user understand the behavior of the developed model through providing explanations regarding why a certain model made certain predictions.[@https://www.aclweb.org/anthology/N16-3020.pdf]

Reviewer: ```suggestion
One way to increase trust in a developed model is by helping users understand the behavior of the developed model through providing explanations regarding why a certain model made certain predictions.[@https://www.aclweb.org/anthology/N16-3020.pdf]
```

Another approach is to provide robust _error analysis_ for newly developed models to help users understand the strengths and weaknesses of a model.[@https://www.aclweb.org/anthology/P19-1073.pdf; @https://www.mitpressjournals.org/doi/abs/10.1162/COLI_a_00072; @doi:10.1093/bioinformatics/bth060]
Adoption of these kind of approaches into biological data and analysis is still rare but is quickly becoming necessary as machine learning approaches become mainstream in biomedicine.

> Finally, methods that can reliably bring disparate datasets together will always remain a need of rare diseases.

Reviewer: ```suggestion
Finally, methods that can reliably integrate disparate datasets will always remain a need of rare diseases.
```

Reviewer: why?

Author: My thoughts are as follows:
1. Since rare disease patients will be sparse by the virtue of rarity of disease...Even if we can ensure that every NF patient sample in US is collected in a standardized manner, patients in UK will be processed differently... but may need to be integrated to have a more substantial dataset.
2. There will always be temporal differences in sample collection and processing, due to the nature of clinical practice. In diseases that are common, samples with significant temporal differences can be disregarded from an analysis, but that luxury is not available to the rare disease field.

> Moreover, combining data that originated from diverse modalities to create a complete picture of the disease related biology is increasingly becoming common.

Reviewer: This seems semi-redundant with above sentence

Author: The above sentence was referring to datasets that are of the same datatype e.g. RNAseq. This is referring to datasets using different datatypes e.g. RNAseq + methylation

This segues to the next "anchor" section.

> To facilitate such analyses in rare disease, methods that rely on finding structural correspondences between datasets ("anchors") may be able to transform the status-quo of using machine learning methods in rare disease.[@https://www.aclweb.org/anthology/W06-1615.pdf; @https://people.cs.umass.edu/~mahadeva/papers/IJCAI2011-DA.pdf; https://www.cell.com/cell/fulltext/S0092-8674(19)30559-8]

Reviewer: This is a complicated topic to introduce as second to last sentence, I suggest removing or moving elsewhere

Author: I agree that it is a complicated topic, but it would be great to have this in the outlook. Happy to move it towards the beginning, or delve a bit deeper into this if everyone is on board.

This is based on a discussion with @jguinney, and I thought it was good food for thought (both for us and the readers) :)

> Overall, we are optimistic about the opportunities of disruptive innovation that exist in applying machine learning in rare disease.

Reviewer: ```suggestion
Overall, we speculate that this an important burgeoning area of research, and we are optimistic about the future of applying machine learning approaches to rare disease.
```

### Outlook

Throughout this perspective, we have highlighted various challenges in applying ML methods to rare disease data as well as examples of approaches that address these challenges.
Scarcity of samples, while significant, is not the only roadblock towards application of ML in rare disease data.
The high dimensionality of modern data requires creative approaches, such as learning new representations of the data, to manage the curse of dimensionality.
It further requires leveraging prior knowledge and transfer learning methods to appropriately interpret data. 

> Additionally, anyone applying machine learning methods on rare disease data is advised to delve into fundamental techniques that increase confidence and penalize complexity of the resultant models to enhance generalizabilty of ML in rare disease.

Reviewer: ```suggestion
Additionally, anyone applying machine learning methods on rare disease data should use techniques that increase confidence (such as bootstrapping) and penalize complexity of the resultant models (such as regularization) to enhance the generalizability of their work. 
```
